modules/vslam/sba.cpp

```diff
@@ -0,0 +1,579 @@
+#include <opencv2/opencv.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+#include <iostream>
+#include <opencv2/core.hpp>
+#include <opencv2/calib3d/calib3d.hpp>
+#include <stdio.h>
+
+using namespace cv;
+#include <vector>
+
+struct CameraParams
+{
+	double ppx = 0.0;
+	double ppy = 0.0;
+	double focal = 0.0;
+	double aspect = 0.0;
+	Mat camera_mat;
+
+	CameraParams(const Mat& camera)
+	{
+		ppx = camera.at<double>(0, 2);
+		ppy = camera.at<double>(1, 2);
+		focal = camera.at<double>(0, 0);
+		aspect = camera.at<double>(1, 1) / camera.at<double>(0, 0);
+		camera_mat = camera.clone();
```

Comment:
Use `Matx33d` to store camera intrinsics.
Avoid cloning values into own variables. Existed OpenCV processing functions accept input as matrix only.
Create "inline" access methods if you need them.